Test/Schema/test_Schema.py

```diff
@@ -18,3 +19,22 @@ def test_load_imageset(dials_data, tmp_path):
         imagesets = load_imagesets("insulin_1_###.img", str(tmp_path))
         assert len(imagesets) == 2
         assert tuple(map(len, imagesets)) == (22, 22)
+
+
+def test_load_reference_geometries(dials_data):
+    """
+    Test `xia2.Schema.load_reference_geometries`.
+
+    Test the function that finds the set of unique instrument models from a list
+    of experiment list files.
+    """
+    files = ["scaled_20_25.expt", "scaled_30.expt", "scaled_35.expt"]
+    files = [(dials_data("l_cysteine_4_sweeps_scaled") / f).strpath for f in files]
+
+    # Check that there are four input instrument models, of which only two are unique.
+    assert (
+        sum([len(ExperimentList.from_file(f, check_format=False)) for f in files]) == 4
```

Comment:
```suggestion
        sum(len(ExperimentList.from_file(f, check_format=False)) for f in files) == 4
```
no explicit list generation required where iterators can be used

Reply:
Yep, quite right.  Thanks.